File: frontend/src/types/issue.ts
export type IssueStatus =
  | "TODO"
  | "IN_PROGRESS"
  | "IN_REVIEW"
  | "DONE";

export type IssuePriority =
  | "LOW"
  | "MEDIUM"
  | "HIGH"
  | "CRITICAL";

export interface IssueAssignee {
  id: string;
  name: string;
  avatar?: string;
}

export interface Issue {
  id: string;
  projectId: string;
  title: string;
  description: string;
  status: IssueStatus;
  priority: IssuePriority;
  assignee?: IssueAssignee;
  labels: string[];
  createdAt: string;
  updatedAt: string;
}

File: backend/app/schemas/issue.py
from pydantic import BaseModel, Field


class Issue(BaseModel):
    id: int
    project_id: int
    assignee_id: int | None
    title: str
    description: str | None
    status: str
    priority: str


class IssueCreate(BaseModel):
    project_id: int
    assignee_id: int | None = None
    title: str = Field(
        min_length=1,
        max_length=200,
    )
    description: str | None = None
    status: str = "TODO"
    priority: str = "MEDIUM"


class IssueUpdate(BaseModel):
    title: str | None = Field(
        default=None,
        min_length=1,
        max_length=200,
    )
    description: str | None = None
    status: str | None = None
    priority: str | None = None
    assignee_id: int | None = None